runtime/drivers/file/watcher.go

```diff
@@ -181,15 +181,15 @@ func (w *watcher) runInner() error {
 				continue
 			}
 
-			if e.Has(fsnotify.Create) {
+			if we.Type == runtimev1.FileEvent_FILE_EVENT_WRITE {
```

Comment:
The `write` and `chmod` events are much more frequent, so calling `os.Stat` every time might be expensive if there are many files or streaming writes (e.g. a file being exported). But maybe it's not a problem.

(Note that this is before the batching in `flush`, so the same file may get checked very frequently here.)

Reply:
Ya I think we can merge events with other events for the path within a batch. That should fix the issue with renamed directories not getting `Dir=true`.